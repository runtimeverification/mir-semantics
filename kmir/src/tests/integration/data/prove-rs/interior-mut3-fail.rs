use std::cell::UnsafeCell;

fn main() {
    let data = UnsafeCell::new(0);
    
    unsafe {
        *data.get() += 42;
    }
    
    assert!(data.into_inner() == 42)
}