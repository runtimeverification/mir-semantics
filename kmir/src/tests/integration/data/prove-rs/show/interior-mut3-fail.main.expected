
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (150 steps)
├─ 3
│   #expect ( thunk ( #applyBinOp ( binOpEq , thunk ( #applyBinOp ( binOpBitAnd , th
│   function: main
┃
┃ (1 step)
┣━━┓
┃  │
┃  ├─ 4
┃  │   AssertError ( assertMessageMisalignedPointerDereference ( ... required: operandC
┃  │   function: main
┃  │
┃  │  (1 step)
┃  └─ 6 (stuck, leaf)
┃      #ProgramError ( AssertError ( assertMessageMisalignedPointerDereference ( ... re
┃      function: main
┃
┗━━┓
   │
   ├─ 5
   │   #execBlockIdx ( basicBlockIdx ( 8 ) ) ~> .K
   │   function: main
   │
   │  (8 steps)
   └─ 7 (stuck, leaf)
       #traverseProjection ( toLocal ( 2 ) , thunk ( #cast ( PtrLocal ( 1 , place ( ...
       function: main
       span: 80


┌─ 2 (root, leaf, target, terminal)
│   #EndProgram ~> .K


